### ISSUE TITLE
Type with Redacted properties should not be allowed to be logged

### ISSUE THREAD

**Issue by LiorBanai** [meziantou/Meziantou.Analyzer#935]
lets say I have the following Type
```csharp
   public partial class PatientInfo
   {
        [PiiData] public string PatientId { get; set; }
        public ulong RecordId { get; set; } //the record Id in the backend database
        [PiiData] public string FirstName { get; set; }
        [PiiData] public string MiddleName { get; set; }
        [PiiData] public string LastName { get; set; }
        public DateTimeOffset? ScheduledDate { get; set; } //UTC value
        public string Department { get; set; }
        public DateTimeOffset? DateOfBirth { get; set; } //UTC value
        [PiiData] public string MedicalRecordNumber { get; set; } //Same as Patient ID
}
```

currently this is detected correctly:
```csharp
// Report the second argument
logger.LogInformation("dummy", obj.PropertyDecoratedWithDataClassificationAttribute);
```

is it possible to prevent logging of the type itself?
```
PatientInfo p = new();
logger.LogInformation("{Patient}", p);
```

since there could be multipe log providers any type that has sensitive attribute should not be logged (since some provder could log inner fields/properties and ignore the attributes). Usage of any member (field, property, parameter) decorated with Microsoft.Extensions.Compliance.Classification.DataClassificationAttribute should not be logged indirectly by logging the object itself

related to https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0153.md

**Comment by meziantou** [comment]
I'm not sure this is the same as properties. When logging a property value directly, you cannot know if the property is decorated with `PiiData`. However, when logging an object, you can find attributes using reflection and redact values properly. I'm not an expert of [data redaction in .NET](https://learn.microsoft.com/en-us/dotnet/core/extensions/data-redaction?tabs=dotnet-cli&WT.mc_id=DT-MVP-5003978), so I may be wrong.

**Comment by LiorBanai** [comment]
the issue is that this is third party implementation (log provider) so better to have analyzer the prevent the logging itself.
in my example, Serilog will log the entire object and if destructor will use even each property will be logged. Better to have ability not to log object and use `Log Properties` that can log object and redact it at the .NET level before the log provider get the data. 

refelction can be use but developers normally just log the types and not aware of sensitive info in the object

**Comment by LiorBanai** [comment]
btw in the documentation (https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0153.md) i think there is typo `Most of the time, these values should not be used with [LogProperties] to redact values.`

it should be `Most of the time, these values should be used with [LogProperties] to redact values.`

**Comment by meziantou** [comment]
The latest version has an option to enable detection of objects containing properties or fields decorated with a data classification attribute

**Comment by LiorBanai** [comment]
Thank you. I'll test it soon

**Comment by LiorBanai** [comment]
the configuration of `MA0153.report_types_with_data_classification_attribute` is not clear. I added it to the project but it does not seems to affect.

[ConsoleApp1.zip](https://github.com/user-attachments/files/24142597/ConsoleApp1.zip)

<img width="1330" height="309" alt="Image" src="https://github.com/user-attachments/assets/e9ed6e9e-8e9a-49d6-be9b-3fcd2a700003" />

**Comment by meziantou** [comment]
I just published the latest version 2.0.259 which include the new configuration flag.

**Comment by LiorBanai** [comment]
the new one does detect it. Thanks

<img width="1236" height="331" alt="Image" src="https://github.com/user-attachments/assets/5355b74f-f1e5-4f85-9108-e2dafbe534b9" />
